File: ruby/api/booking.rb
require 'rest-client'

module Booking

  def all_bookings
    RestClient.get 'https://restful-booker.herokuapp.com/booking'
  rescue => e
    e.response
  end

  def specific_booking(id, accept)
    RestClient.get "https://restful-booker.herokuapp.com/booking/#{id}", accept: accept
  rescue => e
    e.response
  end

  def create_booking(payload, content_type)
    RestClient.post 'https://restful-booker.herokuapp.com/booking', payload, accept: :json, content_type: content_type
  rescue => e
    e.response
  end

  def delete_booking(id, token)
    RestClient.delete "https://restful-booker.herokuapp.com/booking/#{id}", cookie: "token=#{token}"
  rescue => e
    e.response
  end

end

File: java/src/test/java/com/restfulbooker/api/api/Booking.java
package com.restfulbooker.api.api;

import com.restfulbooker.api.payloads.request.BookingPayload;
import com.restfulbooker.api.payloads.response.BookingResponse;
import org.springframework.http.*;
import org.springframework.web.client.HttpClientErrorException;
import org.springframework.web.client.RestTemplate;

import java.util.Arrays;

public class Booking {

    private static RestTemplate restTemplate = new RestTemplate();
    private static String baseUrl = "http://localhost:3001";

    public static ResponseEntity<String> getBookings(){
        return restTemplate.getForEntity(baseUrl + "/booking", String.class);
    }

    public static ResponseEntity<String> getBooking(int id, MediaType accept) throws HttpClientErrorException {
        HttpHeaders requestHeaders = new HttpHeaders();
        requestHeaders.setAccept(Arrays.asList(accept));

        HttpEntity<String> httpEntity = new HttpEntity<String>(requestHeaders);

        return restTemplate.exchange(baseUrl + "/booking/" + Integer.toString(id), HttpMethod.GET, httpEntity, String.class);
    }

    public static ResponseEntity<BookingResponse> postBooking(BookingPayload payload) {
        HttpHeaders requestHeaders = new HttpHeaders();
        requestHeaders.setContentType(MediaType.APPLICATION_JSON);
        requestHeaders.setAccept(Arrays.asList(MediaType.APPLICATION_JSON));

        HttpEntity<BookingPayload> httpEntity = new HttpEntity<BookingPayload>(payload, requestHeaders);

        return restTemplate.exchange(baseUrl + "/booking", HttpMethod.POST, httpEntity, BookingResponse.class);
    }

    public static ResponseEntity<String> deleteBooking(int id, String tokenValue) {
        HttpHeaders requestHeaders = new HttpHeaders();
        requestHeaders.set("Cookie","token=" + tokenValue);

        HttpEntity<String> httpEntity = new HttpEntity<String>(requestHeaders);

        return restTemplate.exchange(baseUrl + "/booking/" + Integer.toString(id), HttpMethod.DELETE, httpEntity, String.class);
    }
}
